### PR TITLE
Add ability to spawn particles only on edges

### PR DESCRIPTION
**Pull request by Supernova4422**
This is particularly useful if a user has a window manager such as yabai with gaps enabled. This change will ensure particles will spawn within only those gaps.

The centre one is included for when two windows are open.

### CODE REVIEW

File: particles.coffee [hunk 1]
```diff
@@ -29,6 +29,13 @@ opts =
   # run at around 60 fps.
   frameSkip: 5
 
+  # Only spawn particles at the edge of the screen.
+  # Particles will only move along the edges.
+  edgeOnly: false
+  # Only useful if edgeOnly is true. When spawning particles, also spawn particles
```
Comment (nepthar) [inline particles.coffee:36]: Could you add a space here to keep this in the same style as above?

File: src/particles.js [hunk 2]
```diff
@@ -74,8 +74,64 @@ class TVar {
 class Particle {
   constructor (net) {
     this.net = net
-    this.x = Math.random() * net.limx
-    this.y = Math.random() * net.limy
+    this.val = Math.random()
```
Comment (nepthar) [inline src/particles.js]: Give this variable a more descriptive name, like "random"

File: src/particles.js [hunk 3]
```diff
@@ -74,8 +74,64 @@ class TVar {
...
+      this.threshold1 = 0.5
+      this.threshold2 = 1.0 
+
+      if (net.opts.edgeAndCentre)
+      {
+        this.threshold1 = 0.3
+        this.threshold2 = 0.6 
+      }
+
+      if (Math.random() <= 0.5)
```
Comment (nepthar) [inline src/particles.js:90]: Maybe just throw in a few helpful comments here like

"top particle", "left particle", "center vertical"... etc.

**Review by nepthar — CHANGES_REQUESTED** [review]
Cool idea! Just a few change suggestions.

Also, I couldn't get the edgeAndCentre option to work. Can you double check that it works for you?

**Comment by nepthar** [comment]
> Cool idea! Just a few change suggestions.
> 
> Also, I couldn't get the edgeAndCentre option to work. Can you double check that it works for you?

Never mind that last part, I was just using the US spelling of 'center'.

**Comment by Supernova4422** [comment]
> 
> 
> Cool idea! Just a few change suggestions.
> 
> Also, I couldn't get the edgeAndCentre option to work. Can you double check that it works for you?

I'll make some edits of this tonight after work. Should be very quick to redo and repackage. Thanks for having a review of it!

**Comment by Supernova4422** [comment]
Changes are implemented. I might've mixed up where the y axis starts from though. Where would be a good place for me to double check?

**Comment by nepthar** [comment]
Well, if it works correctly, it seems okay to me! If you do find a mistake later, feel free to send another quick PR over.
Thanks!

**Pull request merged**